### ISSUE TITLE
Can't install wok package on Ubuntu 14.04

### ISSUE THREAD

**Issue by romu70** [kimchi-project/gingerbase#43]
I installed the dependencies listed in the wok installation documentation, but when I run `sudo dpkg -i wok-*.deb`, I get:

```
(Reading database ... 208299 files and directories currently installed.)
Preparing to unpack wok-2.0.0-0.noarch.deb ...
Unpacking wok (2.0.0) over (2.0.0) ...
Setting up wok (2.0.0) ...
dpkg: error processing package wok (--install):
 subprocess installed post-installation script returned error exit status 127
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Errors were encountered while processing:
 wok
```

A bit cryptic for me, any idea? Thanks.

**Comment by andreas-aeschlimann** [comment]
Same issue here. I have worked with Kimchi <2 before and wanted to upgrade now.

I tried on a clean Ubuntu Server 14.04LTS host, installed all necessary dependencies and executed:

```
sudo dpkg -i wok-*.deb ginger-base-*.deb
Selecting previously unselected package wok.
(Reading database ... 83108 files and directories currently installed.)
Preparing to unpack wok-2.0.0-0.noarch.deb ...
Unpacking wok (2.0.0) ...
Selecting previously unselected package ginger-base.
Preparing to unpack ginger-base-2.0.0-0.noarch.deb ...
Unpacking ginger-base (2.0.0) ...
Setting up wok (2.0.0) ...
dpkg: error processing package wok (--install):
 subprocess installed post-installation script returned error exit status 127
dpkg: dependency problems prevent configuration of ginger-base:
 ginger-base depends on wok; however:
  Package wok is not configured yet.

dpkg: error processing package ginger-base (--install):
 dependency problems - leaving unconfigured
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Errors were encountered while processing:
 wok
 ginger-base
```

**Comment by danielhb** [comment]
@alinefm do we support wok running in ubuntu 14.04 LTS?

**Comment by andreas-aeschlimann** [comment]
@danielhb tried on Debian 8.3 as well, I get the same problem. It would be nice if there were a clean "installation instruction" out there, because right now it is a bit hard to understand for people who have no deep insight in the project (my opinion).

Meanwhile I'll try some other approaches. :-)

**Comment by romu70** [comment]
+1000 for clean installation instructions.

I finally managed to install Kimchi on Ubuntu 15.10 but with some new issues, I'll raise a new ticket.

**Comment by alinefm** [comment]
HI @andreasfcb! Hi @romu70!

I am not sure on what you meant about "clean installation instructions".
You should start reading the README file from Wok. You can find it at: https://github.com/kimchi-project/wok/blob/master/docs/README.md

There is a special section for Linux support, maybe it worth a reading too: https://github.com/kimchi-project/wok/blob/master/docs/README.md#linux-support

As you will see, Kimchi community makes an effort to test Wok and its official plugins in the latest version of Fedora, Ubuntu, RHEL and openSUSE. Which means: Fedora 23, Ubuntu 15.10, RHEL 7.2 and openSUSE 42.1

AFAIK there are some people trying to run Wok and its plugins on Debian and they are facing some issues due the dependencies. There is a bunch of issues open for it.

I suggest to use Ubuntu 15.10, if you want to run on a debian like distribution.

Each plugin has its own README file. Please, take a time to read the installation and troubleshooting sections:

Kimchi: https://github.com/kimchi-project/kimchi/blob/master/docs/README.md
Ginger Base: https://github.com/kimchi-project/gingerbase/blob/master/docs/README.md
Ginger: https://github.com/kimchi-project/ginger/blob/master/docs/README.md

Hope it helps you.
Please, let me know on any issue you have.
You can also join the #kimchi channel on OFTC server for a faster communication.

**Comment by romu70** [comment]
@alinefm, just an example: when I installed Kimchi, I first installed Wok and GingerBase from sources and afterwards, I saw some .deb are available. The available of deb files is not emphasized enough compared to the installation from sources. But perhaps, that may be what you want.

Having a single page "Install on Ubuntu" with the different steps GingerBase, then Wok, then Kimchi, with all instructions in a single place would be a huge improvement.

And having the supported versions would also be great.

**Comment by andreas-aeschlimann** [comment]
Hi @alinefm, I appreciate your answer. I really love Kimchi and I hope I can get it to work, but this time I am really stuck. :-/

To your question. I think the installation instructions are very confusing. For example, it says to install Wok first, but there are no packages/links listed. So I just picked the zip from the "release" tab in the github project. Then, after moving to ginger/gingerbase, it suddenly says to download the *.deb files. This is all very weird and I think the ReadMe would be easier to understand if it just exactly said what to download and where to execute/extract it.

Now back to the issues: So yes, I have read all instructions carefully and installed all dependencies one after each other. I didn't get it to work on Ubuntu 14.04LTS, 15.10 and Debian 8.3 properly (yet). I have to admit I am far from being an expert, so when instructions are not clear I might be confused.

As you suggested, I installed Ubuntu 15.10 now. Installing from the given *.deb packages in the ginger installation ReadMe, I was able to finally connect to `https://myhost.com:8001/`. But I can only see the tab "Host" and no "Virtualization". So basically, I can do nothing. What does that mean? @romu70 how did you do it?

---

For the reference, find here some protocol of my struggles when going through the README's. Maybe you will understand why it is very hard to follow. As you suggested, I tried to install it on Ubuntu 15.10. Here is what I have done:
- Follow Wok instructions, install Debian/Ubuntu dependencies

```
apt-get install gcc make autoconf automake gettext git pkgconf xsltproc logrotate
apt-get install python-cherrypy3 python-cheetah python-pam python-m2crypto python-jsonschema  python-psutil python-ldap python-lxml nginx openssl fonts-font-awesome texlive-fonts-extra
apt-get install g++ python-dev python-pip
pip install cython libsass
apt-get install pep8 pyflakes python-requests
```
- Created folder `/usr/share/wok/`, unzipped into it from this file: `https://github.com/kimchi-project`

```
/wok/archive/2.0.0.zip
./autogen.sh --system
make
```
- So far no errors. Executing `python src/wokd` returns this:
  
  > Traceback (most recent call last):
  >   File "src/wokd", line 101, in <module>
  >     sys.exit(main(sys.argv[1:]))
  >   File "src/wokd", line 98, in main
  >     wok.server.main(options)
  >   File "/usr/share/wok/src/wok/server.py", line 208, in main
  >     srv = Server(options)
  >   File "/usr/share/wok/src/wok/server.py", line 62, in **init**
  >     start_proxy(options)
  >   File "/usr/share/wok/src/wok/proxy.py", line 99, in start_proxy
  >     _create_proxy_config(options)
  >   File "/usr/share/wok/src/wok/proxy.py", line 70, in _create_proxy_config
  >     with open(cert, "w") as f:
  > IOError: [Errno 2] No such file or directory: '/etc/wok/wok-cert.pem'
- Starting to install dependencies for Kimchi, Gingerbase and Ginger

```
apt-get install gcc make autoconf automake gettext git pkgconf xsltproc
apt-get install python-configobj websockify novnc python-libvirt libvirt-bin nfs-common qemu-kvm python-parted python-ethtool sosreport python-ipaddr  python-lxml open-iscsi python-guestfs libguestfs-tools spice-html5 python-magic python-paramiko
apt-get install g++ python-dev python-pip
pip install cython libsass
apt-get install pep8 pyflakes python-requests python-mock
apt-get install python-apt sosreport python-configobj python-parted
apt-get install  libvirt-bin python-libvirt hddtemp python-libuser python-ethtool python-augeas python-ipaddr python-magic python-netaddr
apt-get install python-mock
```
- Unzipped `https://github.com/kimchi-project/kimchi/archive/2.0.0.zip` into `/usr/share/wok/src/wok/plugins/kimchi/`
- Typed the following:

```
./autogen.sh --system
make
```
- Now installing gingerbase. But now, I get instructions to download packages. But what now with the steps I have done before? Ignore the wok package and only install the gingerbase package? Anyway, I proceeded with my current "manual" procedure:
- Unzipped `https://github.com/kimchi-project/gingerbase/archive/2.0.0.zip` into `/usr/share/wok/src/wok/plugins/gingerbase/`
- Typed the following:

```
./autogen.sh --system
make
```
- Unzipped `https://github.com/kimchi-project/ginger/archive/1.3.0.1.zip` into `/usr/share/wok/src/wok/plugins/ginger/`
- Typed the following:

```
./autogen.sh --system
make
```
- Until here fine again. Executing `python src/wokd` returns this:
  
  > Traceback (most recent call last):
  >   File "./src/wokd", line 101, in <module>
  >     sys.exit(main(sys.argv[1:]))
  >   File "./src/wokd", line 98, in main
  >     wok.server.main(options)
  >   File "/usr/share/wok/src/wok/server.py", line 208, in main
  >     srv = Server(options)
  >   File "/usr/share/wok/src/wok/server.py", line 62, in **init**
  >     start_proxy(options)
  >   File "/usr/share/wok/src/wok/proxy.py", line 99, in start_proxy
  >     _create_proxy_config(options)
  >   File "/usr/share/wok/src/wok/proxy.py", line 80, in _create_proxy_config
  >     with open(os.path.join(nginx_config_dir, "wok.conf.in")) as template:
  > IOError: [Errno 2] No such file or directory: '/etc/nginx/conf.d/wok.conf.in'
- And that's where I am lost. 
- As I didn't know what to do, I just tried using the packages provided in the ginger ReadMe. When I just try this here, it works on Ubuntu 15.10 (note: didn't work on Ubuntu 14.04LTS and Debian 8.3 for me):

```
wget http://kimchi-project.github.io/wok/downloads/wok-2.0.0-0.noarch.deb
wget http://kimchi-project.github.io/gingerbase/downloads/ginger-base-2.0.0-0.noarch.deb
wget http://kimchi-project.github.io/ginger/downloads/ginger-2.0.0-0.noarch.deb
dpkg -i wok-*.deb ginger-*.deb
```

As a regular user, I am asking myself. What's with Kimchi? Is there a package as well? Do I have to install it manually? (Yes, I did that.)

**Comment by romu70** [comment]
Hi @andreasfcb, I wrote a simple script to install Kimchi on Ubuntu 15.10, here it is: https://github.com/kimchi-project/kimchi/wiki/Setup-script-for-Ubuntu-15.10. Enjoy!

**Comment by andreas-aeschlimann** [comment]
@romu70 works like a charm, thank you!

P.S. I think you mean Ubuntu 14.04, not 16.04?

**Comment by romu70** [comment]
I couldn't install on 14.04 because of unsatisfied dependencies. I switched to 15.10 and now it works. The script helps a bit to ease the procedure.

I understood 14.04 is not supported. I close the ticket.
